React/NextJs/ProjectStructure.md

```diff
@@ -0,0 +1,212 @@
+
+In this case, inside a specific molecule, we could add a subfolder `layouts` (not to be confused with actual layout described below) to define how our molecule would look like on specific media query.
+
+```
+.
+└── molecules
+    └── Card
+        ├── layouts
+        │   ├── Desktop.tsx
+        │   └── Mobile.tsx
+        └── index.tsx
+```
+
+For this case, inside `index.tsx` we would have something like this:
+```tsx
+  export const Card = () => {
+    return (
+      <Media at="sm">
+        <Mobile />
+      </Media>
+      <Media at="md">
+        <Desktop />
+      </Media>
+    )
+  }
+```
+
+This is the case **only** for layouts that would add too much complexity when using standard css queries.
+
+
+### Layouts
+
+With Layouts, we generally define overall page structure that will be presented to the user based on a specific auth state or a specific page.
+
+For example, your app could have an Admin dashboard on `/admin` route which has a fixed header and sidebars on both right and left, and scrollable content in the middle (you're most likely familiar with this kind of layout).
+
+You will define your different layouts inside `layouts` folder:
+
+```
+.
+.
+└── components
+    └── layouts
+        ├── AdminLayout
+        │   └── index.tsx
+        ├── MainLayout
+        │   └── index.tsx
+        └── BlogLayout
+            └── index.tsx
+```
+
+Then, when creating your routes (pages), you will wrap your page in the layout that represents the current page:
+
+```tsx
+// pages/index.tsx
+
+export default function Home() {
+  return (
+    <MainLayout>
+      ...
+    </MainLayout>
+  )
+}
+
+// pages/admin.tsx
+export default function Admin() {
+  return (
+    <AdminLayout>
+      ...
+    </AdminLayout>
+  )
+}
+```
+
+## Setting up store
+
+Your datx store will be placed in the root of the `src` folder as follows:
+
+```
+src
+└── store
+    ├── models
+    │   ├── User.ts
+    │   └── Session.ts
+    ├── utlis
+    │   ├── config.ts
+    │   └── initializeStore.ts
+    └── index.ts
+```
+
+## Fetchers
+
+In `src/fetchers` you will organize your (swr) fetchers, usually by the model on which you will make API calls.
+
+```ts
+// src/fetchers/user.ts
+
+import { AppCollection } from '../store';
+import { User } from '../store/models/User';
+
+export async function fetchUser(store: AppCollection, id: string): Promise<User> {
+  try {
+    const response = await store.fetch(User, id, {
+      include: ['albums']
+    });
+
+    return response.data as User;
+  } catch(response) {
+    // handle response
+  }
+}
+
+```
+
+You will use your fetcher functions with `useSwr` hook inside of your components (organisms, molecules, pages).
+
+
+## Setting up theming and styles
+
+_Sync with @mamihi_official_
+* Styles will probably be a part of components organization
+
+## Tests
+
+(For PR Review):
```

Comment:
I would also add info about folder structure for tests for other units as well, like store, fetchers, etc.

We should also write some "recipes" about actually writing these tests for various types of units (components, hooks, store, fetchers, etc.). But those details are some other markdown file, perhaps `React/NextJs/Testing.md, this file is just for structure.

Reply:
I agree, but this is maybe a topic for a different section/PR. It should be somewhere in between this one and https://github.com/infinum/frontend-handbook/pull/60